graylog2-web-interface/src/components/event-notifications/event-notification-types/LegacyNotificationForm.jsx

```diff
@@ -0,0 +1,103 @@
+import lodash from 'lodash';
+
+import { Select } from 'components/common';
+import { ConfigurationFormField } from 'components/configurationforms';
+
+class LegacyNotificationForm extends React.Component {
+  static propTypes = {
+    config: PropTypes.object.isRequired,
+    onChange: PropTypes.func.isRequired,
+    legacyTypes: PropTypes.object.isRequired,
+  };
+
+  propagateMultiChange = (newValues) => {
+    const { config, onChange } = this.props;
+    const nextConfig = lodash.cloneDeep(config);
+
+    Object.entries(newValues).forEach((keyAndValue) => {
+      const [key, value] = keyAndValue;
+      nextConfig[key] = value;
+    });
+
+    onChange(nextConfig);
+  };
+
+  propagateChange = (key, value) => {
+    const { config } = this.props;
+    const nextConfiguration = lodash.cloneDeep(config.configuration);
+    nextConfiguration[key] = value;
+    this.propagateMultiChange({ configuration: nextConfiguration });
+  };
+
+  formatLegacyTypes = (legacyTypes) => {
+    return Object.keys(legacyTypes)
+      .map(typeName => ({ label: `Legacy ${legacyTypes[typeName].name}`, value: typeName }));
+  };
+
+  handleSelectNotificationChange = (nextLegacyNotificationType) => {
+    this.propagateMultiChange({
+      callback_type: nextLegacyNotificationType,
+      configuration: {},
+    });
+  };
+
+  handleFormFieldChange = (key, value) => {
+    this.propagateChange(key, value);
+  };
+
+  renderNotificationForm(config, legacyType) {
+    const { configuration } = legacyType;
+
+    const configFields = Object.keys(configuration).map((configKey) => {
+      const configField = configuration[configKey];
+      const configValue = config.configuration[configKey] || configField.default_value;
+
+      return (
+        <ConfigurationFormField typeName={config.callback_type}
+                                configField={configField}
+                                configKey={configKey}
+                                configValue={configValue}
+                                onChange={this.handleFormFieldChange} />
+      );
+    });
+
+    return (
+      <fieldset>
+        {configFields}
+      </fieldset>
+    );
+  }
+
+  render() {
+    const { config, legacyTypes } = this.props;
+    const callbackType = config.callback_type;
+
+    return (
+      <React.Fragment>
+        <fieldset>
+          <FormGroup controlId="notification-legacy-select">
+            <ControlLabel>Choose Legacy Notification</ControlLabel>
+            <Select id="notification-legacy-select"
+                    matchProp="label"
+                    placeholder="Select Legacy Notification"
+                    onChange={this.handleSelectNotificationChange}
+                    options={this.formatLegacyTypes(legacyTypes)}
+                    value={callbackType} />
+            <HelpBlock>Select a Legacy Notification to use on this Event Definition.</HelpBlock>
+          </FormGroup>
+        </fieldset>
+
+        <Alert bsStyle="danger">
```

Comment:
This needs to have some margin at the bottom.

<img width="318" alt="Screenshot 2019-07-19 at 14 43 07" src="https://user-images.githubusercontent.com/716185/61536840-e2688500-aa35-11e9-82d4-b8928ef5162c.png">

Reply:
:heavy_check_mark: